src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/GerritServer.java

```diff
@@ -914,14 +915,21 @@ private void rename(String newName) {
      * @param oldName the old name of the Gerrit server
      */
     private void changeSelectedServerInJobs(String oldName) {
-        for (AbstractProject job : PluginImpl.getConfiguredJobs_(oldName)) {
-            GerritTrigger trigger = (GerritTrigger)job.getTrigger(GerritTrigger.class);
+        for (Job job : PluginImpl.getConfiguredJobs_(oldName)) {
+
+            // TODO: find a way to handle add/remove of triggers
+            if (!(job instanceof AbstractProject)) {
+                return;
```

Comment:
I think that is fine myself. I would rather not block trigger support just
because we are missing a relatively minor feature.
On Jul 27, 2015 3:42 AM, "Tom Fennelly" <notifications@github.com> wrote:

> In
> src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/GerritServer.java
> <https://github.com/jenkinsci/gerrit-trigger-plugin/pull/240#discussion_r35523862>
> :
>
> > @@ -914,14 +915,21 @@ private void rename(String newName) {
> >       * @param oldName the old name of the Gerrit server
> >       */
> >      private void changeSelectedServerInJobs(String oldName) {
> > -        for (AbstractProject job : PluginImpl.getConfiguredJobs_(oldName)) {
> > -            GerritTrigger trigger = (GerritTrigger)job.getTrigger(GerritTrigger.class);
> > +        for (Job job : PluginImpl.getConfiguredJobs_(oldName)) {
> > +
> > +            // TODO: find a way to handle add/remove of triggers
> > +            if (!(job instanceof AbstractProject)) {
> > +                return;
>
> @jacob-keller <https://github.com/jacob-keller> Right, I didn't see a way
> of doing it for workflow. So, the easiest thing (for now) might be to mark
> it as a missing feature for Workflow jobs and not have that point block
> this PR?
>
> Then, separately have a discussion on the merits/demerits etc of
> auto-removing triggers. This will either result in a task to remove the
> feature completely (for all build types), or a task to find a way of making
> it work for workflow.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/gerrit-trigger-plugin/pull/240/files#r35523862>
> .
>